SyncExtension/UseCases/Files/CreateFileUseCase.swift

```diff
@@ -124,4 +158,57 @@ struct CreateFileUseCase {
         
         return progress
     }
+    
+    func generateAndUploadThumbnail(driveItemId: Int, fileURL: URL, destinationURL: URL, encryptedThumbnailDestination: URL) async -> CreateThumbnailResponse? {
+        do {
+            let dimensions = CGSize(width: 512, height: 512)
```

Comment:
These dimensions do not match with all the sizes & resolutions. Also, in drive-web seems to be used frequently 300x300, should we establish a default?

Reply:
Adapted this and agreed with @aboredvaro to use a max 256px size for the largest size, generated thumbnail will maintain the aspect ratio